packages/app-mobile/plugins/PluginRunner/dialogs/hooks/useDialogSize.ts

```diff
@@ -1,26 +1,49 @@
 import useAsyncEffect from '@joplin/lib/hooks/useAsyncEffect';
-import { useState } from 'react';
+import { useRef, useState } from 'react';
 import { DialogContentSize, DialogWebViewApi } from '../../types';
+import { PixelRatio } from 'react-native';
 
 interface Props {
 	dialogControl: DialogWebViewApi;
 	webViewLoadCount: number;
+	watchForSizeChanges: boolean;
 }
 
 const useDialogSize = (props: Props) => {
 	const { dialogControl, webViewLoadCount } = props;
 
 	const [dialogSize, setDialogSize] = useState<DialogContentSize|null>(null);
+	const lastSizeRef = useRef(dialogSize);
+	lastSizeRef.current = dialogSize;
 	useAsyncEffect(async event => {
 		if (!dialogControl) {
 			// May happen if the webview is still loading.
 			return;
 		}
 
-		const contentSize = await dialogControl.getContentSize();
-		if (event.cancelled) return;
+		while (!event.cancelled) {
+			const contentSize = await dialogControl.getContentSize();
+			if (event.cancelled) return;
 
-		setDialogSize(contentSize);
+			const lastSize = lastSizeRef.current;
+			if (contentSize.width !== lastSize?.width || contentSize.height !== lastSize?.height) {
+				// We use 1000 here because getPixelSizeForLayoutSize is guaranteed to return
+				// an integer.
+				const pixelToDpRatio = 1000 / PixelRatio.getPixelSizeForLayoutSize(1000);
+				setDialogSize({
+					width: contentSize.width * pixelToDpRatio,
+					height: contentSize.height * pixelToDpRatio,
+				});
+			}
+
+			if (!props.watchForSizeChanges) return;
+
+			await new Promise<void>(resolve => {
+				setTimeout(() => {
+					resolve();
+				}, 500);
```

Comment:
This polling-based solution is currently based on the resize logic on desktop:

https://github.com/laurent22/joplin/blob/c10e617870523e4734aee36ac90e4240496a0eaf/packages/app-desktop/services/plugins/hooks/useContentSize.ts#L51-L54

Consider replacing this with a [`ResizeObserver`](https://developer.mozilla.org/en-US/docs/Web/API/ResizeObserver)-based solution.